viewer/packages/camera-manager/src/Flexible/FlexibleControls.ts

```diff
@@ -355,26 +392,78 @@ export class FlexibleControls extends EventDispatcher<FlexibleControlsEvent> {
         break;
     }
   };
-  private readonly onMouseDown = (event: PointerEvent) => {
+  private readonly onMouseDown = async (event: PointerEvent) => {
     if (!this.isEnabled) return;
     this._cameraVector.synchronizeEnd();
     switch (event.button) {
       case MOUSE.LEFT: {
-        this.startMouseRotation(event);
+        await this.startMouse(event, false);
         break;
       }
-
       case MOUSE.RIGHT: {
         event.preventDefault();
-        this.startMousePan(event);
+        await this.startMouse(event, true);
         break;
       }
-
       default:
         break;
     }
   };
 
+  private async startMouse(initialEvent: PointerEvent, isRight: boolean): Promise<void> {
+    let prevPosition = getMousePosition(this._domElement, initialEvent.clientX, initialEvent.clientY);
+    let rotator: FlexibleControlsRotator | undefined;
+    let translator: FlexibleControlsTranslator | undefined;
+
+    const onMouseMove = async (event: PointerEvent) => {
+      const position = this.getMousePosition(event);
+      const deltaPosition = position.clone().sub(prevPosition);
+      if (deltaPosition.x == 0 && deltaPosition.y == 0) {
+        return;
+      }
+      if (this._keyboard.isShiftPressed()) {
```

Comment:
The `onMouseMove` function is quite long and does a lot of different things depending on the conditions. It might be beneficial to split it into smaller functions, like
`private async handleMouseMove(event: PointerEvent, deltaPosition: Vector2, isRight: boolean, translator: FlexibleControlsTranslator | undefined, rotator: FlexibleControlsRotator | undefined): Promise<void> {`
` .......`
`}`

Reply:
I have made it smaller now. The rotator is gone, so now it is reasonable size